src/kifshare/anon_controllers.clj

```diff
@@ -0,0 +1,57 @@
+(ns kifshare.anon-controllers
+  (:use [kifshare.config :only [jargon-config irods-anonymous-user]]
+        [clojure-commons.error-codes])
+  (:require [kifshare.ranges :as ranges]
+            [cheshire.core :as cheshire]
+            [clojure.tools.logging :as log]
+            [clj-jargon.init :as jinit]
+            [clj-jargon.item-info :as info]
+            [clj-jargon.item-ops :as ops]
+            [clj-jargon.permissions :as perms]
+            [clojure-commons.file-utils :as ft]
+            [kifshare.errors :as errors]))
+
+(defn- validation-info
+  [cm filepath]
+  (log/info "Validating anonymous access to" filepath)
+  (cond
+    (not (info/exists? cm filepath))
+    :not-exists
+
+    (not (info/is-file? cm filepath))
+    :not-file
+
+    (not (perms/is-readable? cm (irods-anonymous-user) filepath))
+    :not-readable
+
+    :else :ok))
+
+(defmacro validated-with-jargon
```

Comment:
This is just `with-jargon` with a set jargon config and which validates that something's a valid thing to serve anonymously. Bundling it makes the other code shorter, but if people are opposed I can split it up so it's not combining two semi-unrelated things.